src/metabase/models/humanization.clj

```diff
@@ -16,75 +19,111 @@
+
+;; :advanced is the default implementation; splits words with cost-based fn
+(defmethod name->human-readable-name :advanced
+  ([s] (name->human-readable-name :advanced s))
+  ([_, ^String s]
+   ;; explode string on hyphens, underscores, spaces, and camelCase
+   (when (seq s)
+     (str/join " " (for [part  (str/split s #"[-_\s]+|(?<=[a-z])(?=[A-Z])")
+                         :when (not (str/blank? part))
+                         word  (dedupe (flatten (infer-spaces part)))]
+                     (capitalize-word word))))))
+
+;; simple replaces hyphens and underscores with spaces
+(defmethod name->human-readable-name :simple
+  ([s] (name->human-readable-name :simple s))
+  ([_, ^String s]
+   ;; explode on hypens, underscores, and spaces
+   (when (seq s)
+     (str/join " " (for [part  (str/split s #"[-_\s]+")
+                         :when (not (str/blank? part))]
+                     (capitalize-word part))))))
+
+;; :none is just an identity implementation
+(defmethod name->human-readable-name :none
+  ([s]   s)
+  ([_ s] s))
 
 
 (defn- custom-display-name?
-  "Is DISPLAY-NAME a custom name that was set manually by the user in the metadata edit screen?"
+  "Is `display-name` a custom name that was set manually by the user in the metadata edit screen?"
   [internal-name display-name]
-  (and (not= display-name (name->human-readable-name:simple   internal-name))
-       (not= display-name (name->human-readable-name:advanced internal-name))))
-
-(defn- re-humanize-names! [model]
-  (doseq [{id :id, internal-name :name, display-name :display_name} (db/select [model :id :name :display_name])
-          :let                                                      [new-display-name (name->human-readable-name internal-name)]
-          :when                                                     (and (not= display-name new-display-name)
-                                                                         (not (custom-display-name? internal-name display-name)))]
-    (log/info (format "Updating display name for %s '%s': '%s' -> '%s'" (name model) internal-name display-name new-display-name))
-    (db/update! model id
-      :display_name new-display-name)))
+  ;; Try all the different implementations of name->human-readable-name. If any (f internal-name) is equal to
+  ;; display-name we can safely assume that display-name was set automatically using that strategy and stop there.
+  ;; Otherwise if none of the strategies match (i.e. `some` returns `nil`) we can assume a custom name was set.
+  (nil? (some #(= display-name (% internal-name))
+              (vals (methods name->human-readable-name)))))
+
+(defn- re-humanize-names!
+  "Update all non-custom display names of all instances of `model` (e.g. Table or Field). To prevent explosions on very
+  large instances this is done in chunks of 10000 IDs at a time"
+  ;; TODO - once Toucan has the new streaming SELECT functions we can use those instead of manually chunking
```

Comment:
👍 any reason the Toucan stuff can't get merged so that you can use it here?

Reply:
Probably could. I will review it today